src/http/mod.rs

```diff
@@ -70,6 +72,33 @@ pub enum LightMethod {
     Put,
 }
 
+/// Enum that allows a user to pass a `Path` or a `File` type to `send_files`
+pub enum FileOrPath {
+    /// Indicates that the `FileOrPath` is a `File`
+    File((File, String)),
+    /// Indicates that the `FileOrPath` is a `Path`
+    Path(PathBuf),
+}
+
+impl From<String> for FileOrPath{
+    fn from(s: String) -> FileOrPath {
+        FileOrPath::Path(PathBuf::from(&s))
+    }
+}
+
+impl<'a> From<&'a str> for FileOrPath {
+    fn from(s: &'a str) -> FileOrPath {
+        FileOrPath::Path(PathBuf::from(s))
+    }
+}
+
+impl<'a> From<(File, &'a str)> for FileOrPath {
+    fn from(f: (File, &str)) -> FileOrPath {
+        let (file, filename) = f;
+        FileOrPath::File((file, filename.to_owned()))
```

Comment:
Should just access the tuplestruct's values directly instead of pattern matching into bindings.